python/private/pypi/index_sources.bzl

```diff
@@ -26,28 +26,43 @@ def index_sources(line):
         line(str): The requirements.txt entry.
 
     Returns:
-        A struct with shas attribute containing a list of shas to download from pypi_index.
+        A struct with shas attribute containing:
+            * `shas` - a list of shas to download from pypi_index.
+            * `version` - the version of the package.
+            * `marker` - the marker as per PEP508 spec.
+            * `requirement` - a requirement line without the marker. This can
+                be given to `pip` to install a package.
```

Comment:
```suggestion
            * `shas` - list[str]; shas to download from pypi_index.
            * `version` - str; version of the package.
            * `marker` - str; the marker expression, as per PEP508 spec.
            * `requirement` - str; a requirement line without the marker. This can
                be given to `pip` to install a package.
```